From 4c2847a27ae224fa6e1ba32d8bfd747c1270224a Mon Sep 17 00:00:00 2001
From: CHEN Xuan <chenxuan@iscas.ac.cn>
Date: Mon, 17 Aug 2026 16:38:34 +0800
Subject: [PATCH] limit default ROCm batch sizes to 8 on riscv64

---
 common/common.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/common/common.h b/common/common.h
index d8a1689..127bfee 100644
--- a/common/common.h
+++ b/common/common.h
@@ -440,8 +440,13 @@ struct ggml_opt_optimizer_params common_opt_lr_pars(void * userdata);
 struct common_params {
     int32_t n_predict             =    -1; // max. number of new tokens to predict, -1 == no limit
     int32_t n_ctx                 =     0; // context size, 0 == context the model was trained with
+#if defined(__riscv)
+    int32_t n_batch               =     8;
+    int32_t n_ubatch              =     8;
+#else
     int32_t n_batch               =  2048; // logical batch size for prompt processing (must be >=32 to use BLAS)
     int32_t n_ubatch              =   512; // physical batch size for prompt processing (must be >=32 to use BLAS)
+#endif
     int32_t n_keep                =     0; // number of tokens to keep from initial prompt
     int32_t n_chunks              =    -1; // max number of chunks to process (-1 = unlimited)
     int32_t n_parallel            =     1; // number of parallel sequences to decode
-- 
2.53.0

