From b946c0e2b03882a58e9074becbff95a9c21493a9 Mon Sep 17 00:00:00 2001
From: Jvle <keke.oerv@isrc.iscas.ac.cn>
Date: Thu, 20 Aug 2026 15:55:25 +0800
Subject: [PATCH] tests: skip XDP test without privileges

Signed-off-by: Jvle <keke.oerv@isrc.iscas.ac.cn>
---
 socket_xdp_linux_test.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/socket_xdp_linux_test.go b/socket_xdp_linux_test.go
index 4f09487..8f8f50d 100644
--- a/socket_xdp_linux_test.go
+++ b/socket_xdp_linux_test.go
@@ -4,6 +4,7 @@
 package netlink
 
 import (
+	"errors"
 	"os"
 	"testing"
 
@@ -13,6 +14,9 @@ import (
 func TestSocketXDPGetInfo(t *testing.T) {
 	xdpsockfd, err := unix.Socket(unix.AF_XDP, unix.SOCK_RAW, 0)
 	if err != nil {
+		if errors.Is(err, unix.EPERM) {
+			t.Skip("AF_XDP socket creation requires privileges unavailable in OBS")
+		}
 		t.Fatal(err)
 	}
 	defer unix.Close(xdpsockfd)
-- 
2.43.0

