diff --git a/crypto/chacha/chacha_enc.c b/crypto/chacha/chacha_enc.c
index f6fbc119885a8..3cd4f0188d35d 100644
--- a/crypto/chacha/chacha_enc.c
+++ b/crypto/chacha/chacha_enc.c
@@ -22,43 +22,43 @@ typedef union {
     u8 c[64];
 } chacha_buf;
 
-# define ROTATE(v, n) (((v) << (n)) | ((v) >> (32 - (n))))
-
-# ifndef PEDANTIC
-#  if defined(__GNUC__) && __GNUC__>=2 && \
-      !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_NO_INLINE_ASM)
-#   if defined(__riscv_zbb) || defined(__riscv_zbkb)
-#    if __riscv_xlen == 64
-#    undef ROTATE
-#    define ROTATE(x, n) ({ u32 ret;                   \
+#define ROTATE(v, n) (((v) << (n)) | ((v) >> (32 - (n))))
+
+#ifndef PEDANTIC
+#if defined(__GNUC__) && __GNUC__ >= 2 && !defined(OPENSSL_NO_ASM) && !defined(OPENSSL_NO_INLINE_ASM)
+#if defined(__riscv_zbb) || defined(__riscv_zbkb)
+#if __riscv_xlen == 64
+#undef ROTATE
+#define ROTATE(x, n) ({ u32 ret;                   \
                         asm ("roriw %0, %1, %2"        \
                         : "=r"(ret)                    \
-                        : "r"(x), "i"(32 - (n))); ret;})
-#    endif
-#    if __riscv_xlen == 32
-#    undef ROTATE
-#    define ROTATE(x, n) ({ u32 ret;                   \
+                        : "r"(x), "i"(32 - (n))); ret; })
+#endif
+#if __riscv_xlen == 32
+#undef ROTATE
+#define ROTATE(x, n) ({ u32 ret;                   \
                         asm ("rori %0, %1, %2"         \
                         : "=r"(ret)                    \
-                        : "r"(x), "i"(32 - (n))); ret;})
-#    endif
-#   endif
-#  endif
-# endif
-
-# define U32TO8_LITTLE(p, v) do { \
-                                (p)[0] = (u8)(v >>  0); \
-                                (p)[1] = (u8)(v >>  8); \
-                                (p)[2] = (u8)(v >> 16); \
-                                (p)[3] = (u8)(v >> 24); \
-                                } while(0)
+                        : "r"(x), "i"(32 - (n))); ret; })
+#endif
+#endif
+#endif
+#endif
+
+#define U32TO8_LITTLE(p, v)     \
+    do {                        \
+        (p)[0] = (u8)(v >> 0);  \
+        (p)[1] = (u8)(v >> 8);  \
+        (p)[2] = (u8)(v >> 16); \
+        (p)[3] = (u8)(v >> 24); \
+    } while (0)
 
 /* QUARTERROUND updates a, b, c, d with a ChaCha "quarter" round. */
-# define QUARTERROUND(a,b,c,d) ( \
-                x[a] += x[b], x[d] = ROTATE((x[d] ^ x[a]),16), \
-                x[c] += x[d], x[b] = ROTATE((x[b] ^ x[c]),12), \
-                x[a] += x[b], x[d] = ROTATE((x[d] ^ x[a]), 8), \
-                x[c] += x[d], x[b] = ROTATE((x[b] ^ x[c]), 7)  )
+#define QUARTERROUND(a, b, c, d) (                  \
+    x[a] += x[b], x[d] = ROTATE((x[d] ^ x[a]), 16), \
+    x[c] += x[d], x[b] = ROTATE((x[b] ^ x[c]), 12), \
+    x[a] += x[b], x[d] = ROTATE((x[d] ^ x[a]), 8),  \
+    x[c] += x[d], x[b] = ROTATE((x[b] ^ x[c]), 7))
 
 /* chacha_core performs 20 rounds of ChaCha on the input words in
  * |input| and writes the 64 output bytes to |output|. */
@@ -92,10 +92,10 @@ static void chacha20_core(chacha_buf *output, const u32 input[16])
 
 #ifdef INCLUDE_C_CHACHA20
 void ChaCha20_ctr32_c(unsigned char *out, const unsigned char *inp, size_t len,
-                      const unsigned int key[8], const unsigned int counter[4])
+    const unsigned int key[8], const unsigned int counter[4])
 #else
 void ChaCha20_ctr32(unsigned char *out, const unsigned char *inp, size_t len,
-                    const unsigned int key[8], const unsigned int counter[4])
+    const unsigned int key[8], const unsigned int counter[4])
 #endif
 {
     u32 input[16];
@@ -104,17 +104,17 @@ void ChaCha20_ctr32(unsigned char *out, const unsigned char *inp, size_t len,
 
     /* sigma constant "expand 32-byte k" in little-endian encoding */
     input[0] = ((u32)ossl_toascii('e')) | ((u32)ossl_toascii('x') << 8)
-               | ((u32)ossl_toascii('p') << 16)
-               | ((u32)ossl_toascii('a') << 24);
+        | ((u32)ossl_toascii('p') << 16)
+        | ((u32)ossl_toascii('a') << 24);
     input[1] = ((u32)ossl_toascii('n')) | ((u32)ossl_toascii('d') << 8)
-               | ((u32)ossl_toascii(' ') << 16)
-               | ((u32)ossl_toascii('3') << 24);
+        | ((u32)ossl_toascii(' ') << 16)
+        | ((u32)ossl_toascii('3') << 24);
     input[2] = ((u32)ossl_toascii('2')) | ((u32)ossl_toascii('-') << 8)
-               | ((u32)ossl_toascii('b') << 16)
-               | ((u32)ossl_toascii('y') << 24);
+        | ((u32)ossl_toascii('b') << 16)
+        | ((u32)ossl_toascii('y') << 24);
     input[3] = ((u32)ossl_toascii('t')) | ((u32)ossl_toascii('e') << 8)
-               | ((u32)ossl_toascii(' ') << 16)
-               | ((u32)ossl_toascii('k') << 24);
+        | ((u32)ossl_toascii(' ') << 16)
+        | ((u32)ossl_toascii('k') << 24);
 
     input[4] = key[0];
     input[5] = key[1];
diff --git a/crypto/chacha/chacha_riscv.c b/crypto/chacha/chacha_riscv.c
index 6721eaa112a9d..c9f17b35d85dc 100644
--- a/crypto/chacha/chacha_riscv.c
+++ b/crypto/chacha/chacha_riscv.c
@@ -41,19 +41,19 @@
 #include "crypto/riscv_arch.h"
 
 void ChaCha20_ctr32_v_zbb_zvkb(unsigned char *out, const unsigned char *inp,
-                               size_t len, const unsigned int key[8],
-                               const unsigned int counter[4]);
+    size_t len, const unsigned int key[8],
+    const unsigned int counter[4]);
 
 void ChaCha20_ctr32_v_zbb(unsigned char *out, const unsigned char *inp,
-                          size_t len, const unsigned int key[8],
-                          const unsigned int counter[4]);
+    size_t len, const unsigned int key[8],
+    const unsigned int counter[4]);
 
 void ChaCha20_ctr32(unsigned char *out, const unsigned char *inp, size_t len,
-                    const unsigned int key[8], const unsigned int counter[4])
+    const unsigned int key[8], const unsigned int counter[4])
 {
     if (len > CHACHA_BLK_SIZE && RISCV_HAS_ZBB() && riscv_vlen() >= 128
-            && (size_t)out % sizeof(size_t) == 0
-            && (size_t)inp % sizeof(size_t) == 0) {
+        && (size_t)out % sizeof(size_t) == 0
+        && (size_t)inp % sizeof(size_t) == 0) {
         if (RISCV_HAS_ZVKB()) {
             ChaCha20_ctr32_v_zbb_zvkb(out, inp, len, key, counter);
         } else {
