--- ceph-20.2.0/src/jaegertracing/opentelemetry-cpp/CMakeLists.txt.orig	2026-04-30 09:35:00.000000000 +0800
+++ ceph-20.2.0/src/jaegertracing/opentelemetry-cpp/CMakeLists.txt	2026-04-30 09:35:00.000000000 +0800
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)

 # See https://cmake.org/cmake/help/v3.3/policy/CMP0057.html required by certain
 # versions of gtest
