From abba42a759b9b80146f327155d605618f33d96ec Mon Sep 17 00:00:00 2001
From: misaka00251 <liuxin@iscas.ac.cn>
Date: Fri, 26 Jun 2026 17:22:58 +0800
Subject: [PATCH] Using system mimalloc shared library

---
 cpp/cmake_modules/ThirdpartyToolchain.cmake | 68 ++++-----------------
 1 file changed, 12 insertions(+), 56 deletions(-)

diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index 665d1b5..04bc728 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -2349,62 +2349,18 @@ if(ARROW_MIMALLOC)
     message(FATAL_ERROR "Can't use mimalloc with ARROW_ENABLE_THREADING=OFF")
   endif()
 
-  message(STATUS "Building (vendored) mimalloc from source")
-  # We only use a vendored mimalloc as we want to control its build options.
-
-  set(MIMALLOC_LIB_BASE_NAME "mimalloc")
-  if(${UPPERCASE_BUILD_TYPE} STREQUAL "DEBUG")
-    set(MIMALLOC_LIB_BASE_NAME "${MIMALLOC_LIB_BASE_NAME}-${LOWERCASE_BUILD_TYPE}")
-  endif()
-
-  set(MIMALLOC_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/mimalloc_ep/src/mimalloc_ep")
-  set(MIMALLOC_INCLUDE_DIR "${MIMALLOC_PREFIX}/include")
-  set(MIMALLOC_STATIC_LIB
-      "${MIMALLOC_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}${MIMALLOC_LIB_BASE_NAME}${CMAKE_STATIC_LIBRARY_SUFFIX}"
-  )
-
-  set(MIMALLOC_C_FLAGS ${EP_C_FLAGS})
-  if(MINGW)
-    # Workaround https://github.com/microsoft/mimalloc/issues/910 on RTools40
-    set(MIMALLOC_C_FLAGS "${MIMALLOC_C_FLAGS} -DERROR_COMMITMENT_MINIMUM=635")
-  endif()
-
-  set(MIMALLOC_CMAKE_ARGS
-      ${EP_COMMON_CMAKE_ARGS}
-      "-DCMAKE_C_FLAGS=${MIMALLOC_C_FLAGS}"
-      "-DCMAKE_INSTALL_PREFIX=${MIMALLOC_PREFIX}"
-      -DMI_INSTALL_TOPLEVEL=ON
-      -DMI_OVERRIDE=OFF
-      -DMI_LOCAL_DYNAMIC_TLS=ON
-      -DMI_BUILD_OBJECT=OFF
-      -DMI_BUILD_SHARED=OFF
-      -DMI_BUILD_TESTS=OFF
-      # GH-47229: Force mimalloc to generate armv8.0 binary
-      -DMI_NO_OPT_ARCH=ON)
-
-  externalproject_add(mimalloc_ep
-                      ${EP_COMMON_OPTIONS}
-                      URL ${MIMALLOC_SOURCE_URL}
-                      URL_HASH "SHA256=${ARROW_MIMALLOC_BUILD_SHA256_CHECKSUM}"
-                      CMAKE_ARGS ${MIMALLOC_CMAKE_ARGS}
-                      BUILD_BYPRODUCTS "${MIMALLOC_STATIC_LIB}")
-
-  file(MAKE_DIRECTORY ${MIMALLOC_INCLUDE_DIR})
-
-  add_library(mimalloc::mimalloc STATIC IMPORTED)
-  set_target_properties(mimalloc::mimalloc PROPERTIES IMPORTED_LOCATION
-                                                      "${MIMALLOC_STATIC_LIB}")
-  target_include_directories(mimalloc::mimalloc BEFORE
-                             INTERFACE "${MIMALLOC_INCLUDE_DIR}")
-  target_link_libraries(mimalloc::mimalloc INTERFACE Threads::Threads)
-  if(WIN32)
-    target_link_libraries(mimalloc::mimalloc INTERFACE "bcrypt.lib" "psapi.lib")
-  endif()
-  add_dependencies(mimalloc::mimalloc mimalloc_ep)
-
-  list(APPEND ARROW_BUNDLED_STATIC_LIBS mimalloc::mimalloc)
-
-  set(mimalloc_VENDORED TRUE)
+  message(STATUS "Using system mimalloc shared library")
+  find_package(mimalloc REQUIRED)
+
+  if(NOT TARGET mimalloc::mimalloc)
+    add_library(mimalloc::mimalloc SHARED IMPORTED)
+    find_library(MIMALLOC_LIB mimalloc)
+    find_path(MIMALLOC_INCLUDE_DIR mimalloc.h)
+    set_target_properties(mimalloc::mimalloc PROPERTIES
+      IMPORTED_LOCATION "${MIMALLOC_LIB}")
+    target_include_directories(mimalloc::mimalloc INTERFACE "${MIMALLOC_INCLUDE_DIR}")
+  endif()
+  set(mimalloc_VENDORED FALSE)
 endif()
 
 # ----------------------------------------------------------------------
-- 
2.54.0

