Description: allow-riscv64-usegles
Author: Han Gao <rabenda.cn@gmail.com>

Last-Update: 2024-06-20

Index: firefox-140.0.4/toolkit/xre/gfxtest/glxtest.cpp
===================================================================
--- firefox-140.0.4.orig/toolkit/xre/gfxtest/glxtest.cpp
+++ firefox-140.0.4/toolkit/xre/gfxtest/glxtest.cpp
@@ -513,7 +513,7 @@ static bool get_egl_gl_status(EGLDisplay dpy,
   PFNGLGETSTRING glGetString =
       cast<PFNGLGETSTRING>(eglGetProcAddress("glGetString"));
 
-#if defined(__arm__) || defined(__aarch64__)
+#if defined(__arm__) || defined(__aarch64__)  || (defined(__riscv) && __riscv_xlen == 64)
   bool useGles = true;
 #else
   bool useGles = false;
