From ce6d162eeb472e73222a599a3aca3d709ed98cd8 Mon Sep 17 00:00:00 2001
From: Chao Liu <chao.liu.riscv@isrc.iscas.ac.cn>
Date: Mon, 5 Jan 2026 22:23:15 +0800
Subject: [PATCH 6/8] target/riscv: update satp_mode to SV48 for rvsp-ref

Signed-off-by: Chao Liu <chao.liu.riscv@isrc.iscas.ac.cn>
---
 target/riscv/cpu.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/target/riscv/cpu.c b/target/riscv/cpu.c
index 65592b2bdd..a49bda83dd 100644
--- a/target/riscv/cpu.c
+++ b/target/riscv/cpu.c
@@ -2189,11 +2189,24 @@ static RISCVCPUProfile RVA23S64 = {
     }
 };
 
+static RISCVCPUProfile RVSP_RVA23S64 = {
+    .u_parent = &RVA23S64,
+    .s_parent = NULL,
+    .name = "rvsp-rva23s64",
+    .misa_ext = 0,
+    .priv_spec = RISCV_PROFILE_ATTR_UNUSED,
+    .satp_mode = VM_1_10_SV48,
+    .ext_offsets = {
+        RISCV_PROFILE_EXT_LIST_END
+    }
+};
+
 RISCVCPUProfile *riscv_profiles[] = {
     &RVA22U64,
     &RVA22S64,
     &RVA23U64,
     &RVA23S64,
+    &RVSP_RVA23S64,
     NULL,
 };
 
@@ -3314,7 +3327,7 @@ static const TypeInfo riscv_cpu_type_infos[] = {
 
     DEFINE_RISCV_CPU(TYPE_RISCV_CPU_RVSP_REF, TYPE_RISCV_BARE_CPU,
         .misa_mxl_max = MXL_RV64,
-        .profile = &RVA23S64,
+        .profile = &RVSP_RVA23S64,
 
         /*
          * ISA extensions
-- 
2.50.1 (Apple Git-155)

