diff --git a/etelemetry/_version.py b/etelemetry/_version.py
index 899efe7..7909f82 100644
--- a/etelemetry/_version.py
+++ b/etelemetry/_version.py
@@ -42,7 +42,7 @@ def get_config():
     cfg = VersioneerConfig()
     cfg.VCS = "git"
     cfg.style = "pep440"
-    cfg.tag_prefix = ""
+    cfg.tag_prefix = "v"
     cfg.parentdir_prefix = ""
     cfg.versionfile_source = "etelemetry/_version.py"
     cfg.verbose = False
diff --git a/setup.cfg b/setup.cfg
index 17f6fb7..bd39e78 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -49,5 +49,5 @@ VCS = git
 style = pep440
 versionfile_source = etelemetry/_version.py
 versionfile_build = etelemetry/_version.py
-tag_prefix =
+tag_prefix = v
 parentdir_prefix =
